systemtap: init at 51b7ca

bendlas opened the pull request:
###### Motivation for this change

Systemtap is very useful in monitoring kernel and userspace execution. Apparently it got removed in january: https://www.mail-archive.com/nix-commits-bounces@lists.science.uu.nl/msg00047.html

I got this version working on kernel 4.7.3, though.
###### Things done
- Built on platform(s)
  - [x] NixOS
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
- [ ] Provide debug info for kernel and userland, in a way accessible to systemtap

---

master needs to be tracked in order to support recent kernels

bendlas committed:
systemtap: init at 51b7ca

master needs to be tracked in order to support recent kernels

joachifm commented on pkgs/development/tools/profiling/systemtap/default.nix:
```diff
@@ -0,0 +1,56 @@
+  sha256 = "18m3lf11r99f4yr1m172548lghc0i22zqbys1fwlwakbczz0a2lz";
+
+  inherit (linuxPackages) kernel;
+  version = kernel.stdenv.lib.substring 0 6 rev;
```
Version strings should satisfy `new > old`. The guidelines now recommend using the commit date in ISO 8601 format.

dezgeg commented on pkgs/development/tools/profiling/systemtap/default.nix:
```diff
@@ -0,0 +1,56 @@
+  rm $out/bin/stap
+  makeWrapper $stapBuild/bin/stap $out/bin/stap \
+    --add-flags "-r $kernelBuildDir" \
+    --prefix PATH : ${kernel.stdenv.cc.cc}/bin:${elfutils}/bin:${gnumake}/bin
```
This could use `lib.makeBinPath`

dezgeg commented on pkgs/top-level/all-packages.nix:
```diff
@@ -14963,6 +14963,8 @@ in
   synergy = callPackage ../applications/misc/synergy { };
 
+  systemtap = callPackage ../development/tools/profiling/systemtap { };
+
```
You need to place this inside the `linuxPackagesFor` function so it can be built on other kernels than the default `linuxPackages`.

groxxda commented on pkgs/development/tools/profiling/systemtap/default.nix:
```diff
@@ -0,0 +1,56 @@
+, elfutils, pkgconfig, gettext, stow, gnumake }:
+let
+  ## fetchgit info
+  url = git://sourceware.org/git/systemtap.git;
```
Please use `https` if possible

bendlas replied:
there only seems to be the `git:` uri available.

groxxda replied:
Okay sorry, I thought it was available since the homepage states it is... :disappointed:

bendlas replied:
I see, there is https://gitlab.com/fche/systemtap.git
it's not as up-to-date as the `git:` uri, that might not matter much for packaging, though

do we prefer `https` because of firewalls?

groxxda commented on pkgs/development/tools/profiling/systemtap/default.nix:
```diff
@@ -0,0 +1,56 @@
+  buildInputs = [ makeWrapper ];
+  meta = with kernel.stdenv.lib; {
+    homepage = https://sourceware.org/systemtap/;
+    repositories.git = git://sourceware.org/git/systemtap.git;
```
Maybe reuse `url` here

groxxda commented on pkgs/development/tools/profiling/systemtap/default.nix:
```diff
@@ -0,0 +1,56 @@
+  meta = with kernel.stdenv.lib; {
+    homepage = https://sourceware.org/systemtap/;
+    repositories.git = git://sourceware.org/git/systemtap.git;
+    description = "SystemTap provides a simple command line interface and scripting language for writing instrumentation for a live running kernel plus user-space applications.";
```
This rather fits`longDescription`.
Convention for `description`: don't include the package name and don't end with a dot

bendlas committed:
systemtap: clean up as per feedback

- shorter description
- move systemtap into linuxPackages
  this is only necessary for the wrapper
  systemtap also has a dyninsn backend to instrument userspace apps
  we could experiment with other platforms, using this
- get rid of stow build dependency
- update version
- minor cleanup steps

groxxda commented:
https://nixos.org/nixpkgs/manual/#sec-sources 

> On 16 Sep 2016, at 22:29, Herwig Hochleitner <notifications@github.com> wrote:
> 
> @bendlas commented on this pull request.
> 
> In pkgs/development/tools/profiling/systemtap/default.nix:
> 
> > @@ -0,0 +1,56 @@
> +{ fetchurl, linuxPackages, makeWrapper, runCommand, fetchgit
> +, elfutils, pkgconfig, gettext, stow, gnumake }:
> +let
> +  ## fetchgit info
> +  url = git://sourceware.org/git/systemtap.git;
> I see, there is https://gitlab.com/fche/systemtap.git
> it's not as up-to-date as the git: uri, that might not matter much for packaging, though
> 
> do we prefer https because of firewalls?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

bendlas commented:
Can this be merged, or is the `git:` source a blocker?

groxxda reviewed (approved):
Since it is the only primary way probided by upstream it's fine 
LGTM
